### PR TITLE
Kashyap/objects create helper conflicts resolved

### PR DESCRIPTION
**Pull request by tangyyyyyyy**
- added ornaments
- added community
- added createObject helper function
- resolved 1000000 merge conflicts

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .env
 node_modules
+testing_behavior.js
```
Comment (missjantastic) [inline .gitignore]: delete?

File: createHelper.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const { Item, itemSchema } = require('./server/schemas/item');
+
+async function createItem(name, rarity, image) {
```
Comment (missjantastic) [inline createHelper.js]: I think this should be in items-service now

File: main.js [hunk 3]
```diff
@@ -9,51 +9,130 @@ const RARITY = {
...
+const { Player }= require('./server/schemas/player');
+const { Creature }= require('./server/schemas/creature');
+
+const { createItem } = require('./server/services/items-service')
+
 const { token, databaseToken } = process.env; 
 const { connect, connection } = require('mongoose');
+const { Community } = require("./server/schemas/community");
+
+const { createItem } = require("./createHelper")
```
Comment (missjantastic) [inline main.js]: createItem is in here twice

Comment (missjantastic) [reply main.js]: remove this one

File: main.js [hunk 4]
```diff
@@ -9,51 +9,130 @@ const RARITY = {
...
     await connect(databaseToken, {dbName: 'christmas_bot_v2'}).catch(console.error);
     console.log("Christmas Bot Status: ONLINE");
 
-    // await createItem('Gumdrop Button', RARITY.COMMON);
-    // await createOrnament('Gumdrop Ornament', RARITY.LEGENDARY, undefined, 123);
-    // await initializeSettings(123);
+
+    //TESTING: set to true if you want to purge databases
+    let input = true
+    if (input) {
```
Comment (missjantastic) [inline main.js]: very nice quality of life improvement 😌

File: main.js [hunk 5]
```diff
@@ -9,51 +9,130 @@ const RARITY = {
...
     } catch(err) {
-        console.log(`Failed to add ornament to the database`);
+        console.log(`Failed to add creature to the database:`);
         console.log(err);
     }
-    
+    return creature
+}
+
+async function createCommunity(serverId, foundOrnaments, foundCreatures) {
```
Comment (missjantastic) [inline main.js]: createCommunity is in here twice

File: server/schemas/community.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+const { Schema, model } = require('mongoose');
```
Comment (missjantastic) [inline server/schemas/community.js:1]: NOICE

File: server/schemas/creature.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+const { Schema, model } = require('mongoose');
```
Comment (missjantastic) [inline server/schemas/creature.js:1]: NOICE

File: server/schemas/item.js [hunk 8]
```diff
@@ -3,7 +3,8 @@ const { Schema, model } = require('mongoose');
 const itemSchema = new Schema({
     name: {
         type: String, 
-        required: true
+        required: true,
+        unique: true
```
Comment (missjantastic) [inline server/schemas/item.js:7]: great way to prevent duplicates in the database 😌

File: server/utils/helper-functions.js [hunk 9]
```diff
@@ -1 +1,15 @@
-// helper functions if we ever need them
\ No newline at end of file
+// helper functions if we ever need themconst { Item, itemSchema } = require('./server/schemas/item');
+
+async function createObject(myObject) {
+    try {
+        await myObject.save()
```
Comment (missjantastic) [inline server/utils/helper-functions.js]: name this something else that's not `myObject`

Comment (missjantastic) [reply server/utils/helper-functions.js]: `objectToCreate`

Comment (missjantastic) [reply server/utils/helper-functions.js]: `createDatabaseEntry` as kashyap suggested

File: createHelper.js
Comment (Awood1029) [inline createHelper.js:1]: What's the difference between this and the createObject() function in helper-functions?

File: main.js [hunk 10]
```diff
@@ -9,51 +9,130 @@ const RARITY = {
...
+
+
 
     await connection.close();
     console.log("Christmas Bot Status: OFFLINE");
 }
 
-async function createItem(name, rarity, image) {
-    const item = new Item({name, rarity, image})
+async function createOrnament(name, rarity, image, serverId, isFound ) {
```
Comment (Awood1029) [inline main.js]: I assume the plan is to eventually move this to a service file like item-service?

File: createHelper.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+    const item = new Item({name, rarity, image})
+    
+    if (item) {
+        await createObject(item);
+        return item
+    }
+
+    return -1;
```
Comment (Awood1029) [inline createHelper.js]: I would probably return null here or something other than -1

File: main.js [hunk 12]
```diff
@@ -9,51 +9,130 @@ const RARITY = {
     LEGENDARY: 0.5,
```
Comment (missjantastic) [inline main.js]: let's move this file to server folder

File: createHelper.js (hunk 2 above)
Comment (missjantastic) [reply createHelper.js]: This file should actually be deleted

File: server/services/items-service.js [hunk 13]
```diff
@@ -1 +1,18 @@
...
+const { createObject } = require('../utils/helper-functions')
+const { Item } = require('../schemas/item')
+
+async function createItem(name, rarity, image) {
+
+    const item = new Item({name, rarity, image})
+    
+    if (item) {
+        await createObject(item);
+        return item
```
Comment (missjantastic) [inline server/services/items-service.js]: we've decided we should delete this and make this function not return anything

File: server/services/items-service.js [hunk 14]
```diff
@@ -1 +1,18 @@
-// createItem()
\ No newline at end of file
+// createItem()
+
+const { createObject } = require('../utils/helper-functions')
+const { Item } = require('../schemas/item')
+
```
Comment (missjantastic) [inline server/services/items-service.js:5]: let's start adding jsdocs to our functions

File: createHelper.js
Comment (tangyyyyyyy) [reply createHelper.js:1]: artifact of the merge conflict disaster of 9/17/2024

this will be deleted

**Review by missjantastic — APPROVED** [review]
BEAUTIFUL WORK

**Pull request merged**